types/react-calendar-timeline/index.d.ts

```diff
@@ -1,131 +1,366 @@
-      sidebarContent?: React.ReactNode;
-      rightSidebarWidth?: number;
-      rightSidebarContent?: React.ReactNode;
-      dragSnap?: number;
-      minResizeWidth?: number;
-      stickyOffset?: number;
-      stickyHeader?: boolean;
-      headerRef?: any;
-      lineHeight?: number;
-      headerLabelGroupHeight?: number;
-      headerLabelHeight?: number;
-      itemHeightRatio?: number;
-      minZoom?: number;
-      maxZoom?: number;
-      clickTolerance?: number;
-      canMove?: boolean;
-      canChangeGroup?: boolean;
-      canResize?: boolean | "left" | "right" | "both";
-      useResizeHandle?: boolean;
-      showCursorLine?: boolean;
-      stackItems?: boolean;
-      traditionalZoom?: boolean;
-      itemTouchSendsClick?: boolean;
-      timeSteps?: TimelineTimeSteps,
-      onItemMove?(itemId: number, dragTime: number, newGroupOrder: number): any;
-      onItemResize?(itemId: number, newResizeEnd: number, edge: "left" | "right"): any;
-      onItemSelect?(itemId: number, e: any, time: number): any;
-      onItemClick?(itemId: number, e: any, time: number): any;
-      onItemDoubleClick?(itemId: number, e: any, time: number): any;
-      onCanvasClick?(groupId:number, time:number, e:any): any;
-      onCanvasDoubleClick?(group: TimelineGroup, time: number, e: any): any;
-      onCanvasContextMenu?(group: TimelineGroup, time: number, e: any): any;
-      onZoom?(timelineContext: TimelineContext): any;
-      moveResizeValidator?(action:"move" | "resize", itemId: number, time: number, resizeEdge: "left" | "right"): any;
-      headerLabelFormats?: TimelineHeaderLabelFormat,
-      subHeaderLabelFormats?: TimelineHeaderLabelFormat,
-      onTimeChange?(visibleTimeStart: number, visibleTimeEnd: number, updateScrollCanvas: (start: number, end: number) => void): any;
-      onTimeInit?(canvasTimeStart: number, canvasTimeEnd: number): any;
-      onBoundsChange?(canvasTimeStart: number, canvasTimeEnd: number): any;
-      itemRenderer?: (props: {item: TimelineItem, context: TimelineContext}) => React.ReactElement<{}>;
-      groupRenderer?: (props: {group: TimelineGroup, isRightSidebar: boolean}) => React.ReactElement<{}>;
-      minimumWidthForItemContentVisibility?: number;
-      children?: any;
-  }
-
-  export const defaultHeaderLabelFormats: TimelineHeaderLabelFormat;
-  export const defaultSubHeaderLabelFormats: TimelineHeaderLabelFormat;
-
-  let ReactCalendarTimeline : React.ClassicComponentClass<ReactCalendarTimelineProps>;
-  export default ReactCalendarTimeline;
-  
+// TypeScript Version: 3.5
+
+import * as React from 'react';
+import { Moment } from 'moment';
+
+declare module 'react-calendar-timeline' {
+    export interface TimelineGroupBase {
+        id: number;
+        title: React.ReactNode;
+        rightTitle?: React.ReactNode;
+    }
+
+    export interface TimelineItemBase<DateType> {
+        id: number;
+        group: number;
+        title?: React.ReactNode;
+        start: DateType;
```

Comment:
@Ilaiwi hey man. One question. Do these should be `start_time` and `end_time` or `start` and `end` to work fine?
Because it stopped working for us with that but in CodeSandbox examples `start` and `end` is used and now I'm wondering what is wrong and what is right.

Reply:
the problem here is that the package provides a "keys" object where you can change what keys you provide for items and groups. 
The issue here that this makes typescript support much harder but I would guess maybe a keys genric should be passed or inferred to the component.

Reply:
Ah gotcha, it's overridable.
![image](https://user-images.githubusercontent.com/3940079/65143764-0c2e3000-da16-11e9-93f2-a37595ee4c8e.png)

IMO it's a bit useless flexibility :) I would rather write some mapping over some data to match it to the shape (what will be needed anyway) than add some odd keys mapping.

Will change it to `start_time` and `end_time`

Reply:
I agree. legacy code is always an issue